sdk/include/opentelemetry/sdk/common/atomic_shared_ptr.h

```diff
@@ -44,7 +44,7 @@ class AtomicSharedPtr
 
   void store(const std::shared_ptr<T> &other) noexcept
   {
-    ptr_.store(new std::shared_ptr<T>(other), std::memory_order_release);
+    delete ptr_.exchange(new std::shared_ptr<T>(other));
```

Comment:
Could we use a mutex to handle the gcc-4.8 case?

This isn't thread-safe since you can have one thread deleting shared_ptr* (from store) while another thread is copying the shared_ptr* (from load)